baseplate/frameworks/thrift/__init__.py

```diff
@@ -41,13 +50,25 @@ def call_with_context(*args: Any, **kwargs: Any) -> Any:
                 # should be expected in the protocol
                 span.finish(exc_info=sys.exc_info())
                 raise
+            except Error as exc:
+                # mark 5xx errors as failures since those are still "unexpected"
+                if exc.code / 100 == 5:
```

Comment:
I think this will only work for 500, right?
```
>>> 504 / 100 == 5
False
```

I think we want integer division:
```
>>> 504 // 100 == 5
True
```

Reply:
also would be easier to read as `if 500 <= exc.code < 600:`